app/controllers/course/announcements_controller.rb

```diff
@@ -0,0 +1,45 @@
+class Course::AnnouncementsController < Course::ModuleController
+  load_and_authorize_resource :announcement, through: :course, class: Course::Announcement.name
+
+  def index #:nodoc:
+    @announcements = @course.announcements.accessible_by(current_ability).sorted
```

Comment:
this should also be handled by Cancancan. I think you only need to do `@announcements = @announcements.sorted`